src/pk/ecc/ecc_dp_find_by_name.c

```diff
@@ -0,0 +1,30 @@
+/* LibTomCrypt, modular cryptographic library -- Tom St Denis
+ *
+ * LibTomCrypt is a library that provides various cryptographic
+ * algorithms in a highly modular and flexible manner.
+ *
+ * The library is free for all purposes without any express
+ * guarantee it works.
+ */
+
+#include "tomcrypt.h"
+
+#ifdef LTC_MECC
+
+ltc_ecc_set_type* ecc_dp_find_by_name(char *curve_name)
+{
+   int i;
+
+   for (i = 0; ltc_ecc_sets[i].size != 0; i++) {
+      if (ltc_ecc_sets[i].name != NULL && XSTRCMP(ltc_ecc_sets[i].name, curve_name) == 0) {
+         return (ltc_ecc_set_type*)&ltc_ecc_sets[i];
```

Comment:
This stuff gets even more complicated in ECC part 2 where we are importing keys in OpenSSL compatible DER format:
* one scenario is that the imported key has curve specified by OID - in this case we use `ecc_dp_find_by_oid(..)` which finds corresponding `ltc_ecc_sets[i]` item (const)
* the other scenario is that the imported key has explicitly defined curve parametes - here we can try whether these curve params do exist in our `ltc_ecc_sets` - this is the idea behind `ecc_dp_find_by_params(..)` - otherwise we have to allocate and properly fill `ltc_ecc_set_type` structure and free it during the key destruction in `ecc_free` (here it is tricky as `ecc_free` has no idea whether the `ltc_ecc_set_type` structure was dynamically allocated or not)